From 181909e2ae8c08fe5f770f23e552268bf9746ecd Mon Sep 17 00:00:00 2001
From: Alan Griffiths <alan@octopull.co.uk>
Date: Mon, 8 Feb 2021 16:15:24 +0000
Subject: [PATCH 06/10] Fix FTBFS.

Missing headers, unnecessary copies, etc.

(cherry picked from commit 418169b058ae84bd729358e7da8ac14b05401006)
---
 examples/example-server-lib/tiling_window_manager.cpp | 1 +
 src/miral/static_display_config.cpp                   | 2 +-
 src/server/frontend_wayland/wl_pointer.h              | 3 ++-
 src/server/input/default_input_manager.h              | 3 ++-
 4 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/examples/example-server-lib/tiling_window_manager.cpp b/examples/example-server-lib/tiling_window_manager.cpp
index 56003abd95..e8ec736edf 100644
--- a/examples/example-server-lib/tiling_window_manager.cpp
+++ b/examples/example-server-lib/tiling_window_manager.cpp
@@ -25,6 +25,7 @@
 
 #include <linux/input.h>
 #include <algorithm>
+#include <limits>
 #include <csignal>
 
 namespace ms = mir::scene;
diff --git a/src/miral/static_display_config.cpp b/src/miral/static_display_config.cpp
index 52f1b8e21b..5ebe9b8424 100644
--- a/src/miral/static_display_config.cpp
+++ b/src/miral/static_display_config.cpp
@@ -145,7 +145,7 @@ try
         if (!cards.IsDefined() || !cards.IsSequence())
             throw mir::AbnormalExit{error_prefix + "invalid 'cards' in '" + ll.first.as<std::string>() + "' layout"};
 
-        for (Node const card : cards)
+        for (Node const& card : cards)
         {
             mg::DisplayConfigurationCardId card_no;
 
diff --git a/src/server/frontend_wayland/wl_pointer.h b/src/server/frontend_wayland/wl_pointer.h
index 5d495246e5..7922412a8f 100644
--- a/src/server/frontend_wayland/wl_pointer.h
+++ b/src/server/frontend_wayland/wl_pointer.h
@@ -25,8 +25,9 @@
 #include "mir/geometry/point.h"
 #include "mir/geometry/displacement.h"
 
-#include <functional>
 #include <chrono>
+#include <functional>
+#include <optional>
 
 struct MirInputEvent;
 typedef unsigned int MirPointerButtons;
diff --git a/src/server/input/default_input_manager.h b/src/server/input/default_input_manager.h
index d4ebd827cf..29ad08a8d6 100644
--- a/src/server/input/default_input_manager.h
+++ b/src/server/input/default_input_manager.h
@@ -21,8 +21,9 @@
 
 #include "mir/input/input_manager.h"
 
-#include <thread>
 #include <atomic>
+#include <memory>
+#include <thread>
 
 namespace mir
 {
-- 
2.34.1

